specs/Workers.md

```diff
@@ -0,0 +1,1228 @@
+  ///
+  /// This corresponds to the `options.name` property passed to the `Worker` constructor in the DOM.
+  /// For more information, see the [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/API/Worker/Worker).
+  ///
+  /// The caller must free the returned string with `CoTaskMemFree`.  See
+  /// [API Conventions](/microsoft-edge/webview2/concepts/win32-api-conventions#strings).
+  [propget] HRESULT Name([out, retval] LPWSTR* value);
+}
+```
+
+```c# (but really MIDL3)
+runtimeclass CoreWebView2DedicatedWorkerCreatedEventArgs;
+runtimeclass CoreWebView2DedicatedWorker;
+runtimeclass CoreWebView2ServiceWorkerRegistration;
+runtimeclass CoreWebView2ServiceWorkerRegisteredEventArgs;
+runtimeclass CoreWebView2ServiceWorkerManager;
+runtimeclass CoreWebView2ServiceWorker;
+runtimeclass CoreWebView2SharedWorkerManager;
+runtimeclass CoreWebView2SharedWorkerCreatedEventArgs;
+runtimeclass CoreWebView2SharedWorker;
+
+namespace Microsoft.Web.WebView2.Core
+{
+    [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2_25")]
+    {
+        event Windows.Foundation.TypedEventHandler<CoreWebView2, CoreWebView2DedicatedWorkerCreatedEventArgs> DedicatedWorkerCreated;
+    }
+
+    // ...
+    runtimeclass CoreWebView2DedicatedWorkerCreatedEventArgs
+    {
+        CoreWebView2DedicatedWorker Worker { get; };
+    }
+
+    runtimeclass CoreWebView2DedicatedWorker
+    {
+        String Name { get; };
+
+        String ScriptUri { get; };
+
+        event Windows.Foundation.TypedEventHandler<CoreWebView2DedicatedWorker, IInspectable> Destroyed;
+    }
+
+    enum CoreWebView2ServiceWorkerState
+    {
+        New = 0,
+        Installing = 1,
+        Installed = 2,
+        Activating = 3,
+        Activated = 4,
+        Redundant = 5,
+    };
+
+    [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2Profile9")]
+    {
+        CoreWebView2ServiceWorkerManager ServiceWorkerManager { get; };
+
+        CoreWebView2SharedWorkerManager SharedWorkerManager { get; };
+    }
+
+    runtimeclass CoreWebView2ServiceWorkerManager
+    {
+        event Windows.Foundation.TypedEventHandler<CoreWebView2ServiceWorkerManager, CoreWebView2ServiceWorkerRegisteredEventArgs> ServiceWorkerRegistered;
+
+        Windows.Foundation.IAsyncOperation<IVectorView<CoreWebView2ServiceWorkerRegistration>> GetServiceWorkerRegistrationsAsync();
+
+        Windows.Foundation.IAsyncOperation<CoreWebView2ServiceWorkerRegistration> GetServiceWorkerRegistrationAsync(String Scope);
+    }
+
+    runtimeclass CoreWebView2ServiceWorkerRegisteredEventArgs
+    {
+        CoreWebView2ServiceWorkerRegistration ServiceWorkerRegistration { get; };
+    }
+
+    runtimeclass CoreWebView2ServiceWorkerRegistration
+    {
+        String Scope { get; };
+
+        event Windows.Foundation.TypedEventHandler<CoreWebView2ServiceWorkerRegistration, IInspectable> Destroyed;
+
+        Windows.Foundation.IAsyncOperation<CoreWebView2ServiceWorker> GetServiceWorkerAsync();
```

Comment:
This gives info on active service worker as host apps in general interested in communicating with the active ones. Do you think we should match with the DOM one's like a method to get installing SW or waiting SW?